### PR TITLE
 fix: Components directory can't be found when multiple settings file are used

### PR DESCRIPTION
**Pull request by bastinc**
Here is a fix for  #290

### CODE REVIEW

File: django_components/template_loader.py [hunk 1]
```diff
@@ -14,15 +14,9 @@ def get_dirs(self):
...
-        if settings.SETTINGS_MODULE:
-            settings_path = Path(*settings.SETTINGS_MODULE.split("."))
-
-            path = (settings_path / ".." / component_dir).resolve()
-            if path.is_dir():
-                directories.append(path)
-
-            path = (settings_path / ".." / ".." / component_dir).resolve()
-            if path.is_dir():
+        if hasattr(settings, "BASE_DIR"):
```
Comment (EmilStenstrom) [inline django_components/template_loader.py:17]: BASE_DIR is not a standardized variable name that contains the base directory, so I don't think we can use it.

File: django_components/template_loader.py [hunk 2]
```diff
@@ -14,15 +14,9 @@ def get_dirs(self):
...
+            path = Path(settings.BASE_DIR, component_dir)
+            if path.exists():
```
Comment (EmilStenstrom) [inline django_components/template_loader.py:19]: You are removing .is_dir() and replacing it with .exists(), is there a reason?

File: django_components/template_loader.py (hunk 1 above)
Comment (bastinc) [reply django_components/template_loader.py:17]: Indeed

File: django_components/template_loader.py [hunk 3]
```diff
@@ -14,15 +14,9 @@ def get_dirs(self):
...
-            settings_path = Path(*settings.SETTINGS_MODULE.split("."))
-
-            path = (settings_path / ".." / component_dir).resolve()
-            if path.is_dir():
-                directories.append(path)
-
-            path = (settings_path / ".." / ".." / component_dir).resolve()
-            if path.is_dir():
+        if hasattr(settings, "BASE_DIR"):
+            path = Path(settings.BASE_DIR, component_dir)
```
[the remark this answers is not in the document]
Comment (bastinc) [reply django_components/template_loader.py:18]: The problem trying to find the components folder from the  settings.SETTINGS_MODULE path is that someone could have a structure more complex (settings folder -> dev folder -> settings.py) it  wouln't work either.

**Comment by EmilStenstrom** [comment]
Does anyone have an idea of how to move forward on this issue? Supporting setups where the settings files are split into many files seems worthwhile, but requiring the use of the non-standard BASE_DIR is not the right solution imho.

**Comment by matevz-ap** [comment]
We also have 2 settings files in our project and had trouble implementing the library, but we figured out that its due to our projects file structure. Our structure looks like this `project > project > settings > settings.py`, so none of the `path = (settings_path / ".." / ".." / component_dir).resolve()` paths can resolve it. Our idea is to maybe add  a new optional variable such as COMPONENT_DIR, that could be specified if the user has a different file structure.

**Comment by EmilStenstrom** [comment]
This is now solved with this setting: https://github.com/EmilStenstrom/django-components#configure-the-module-where-components-are-loaded-from

Let me know if this doesn't solve this problem, and we'll reopen.

**Pull request closed without merging**